lib/components/map/bounds-updating-overlay.js

```diff
@@ -69,7 +69,31 @@ class BoundsUpdatingOverlay extends MapLayer {
     const oldIntermediate = oldProps.query && oldProps.query.intermediatePlaces
     const newIntermediate = newProps.query && newProps.query.intermediatePlaces
     const intermediateChanged = !isEqual(oldIntermediate, newIntermediate)
-    if (
+
+    // Also refit map if itineraryView prop has changed.
+    const itineraryViewChanged = oldProps.itineraryView !== newProps.itineraryView
+
+    if (itineraryViewChanged) {
+      // If itineraryView has changed,
+      // force a resize of the map before re-fitting the active itinerary or active leg,
+      // and do that after a delay to that canvas heights have stabilized in the DOM.
```

Comment:
```suggestion
      // and do that after a delay to ensure that canvas heights have stabilized in the DOM.
```

Reply:
Updated in https://github.com/opentripplanner/otp-react-redux/pull/339/commits/794aeafcdf59f7ea394a00aac565daa4e7a9b2f0.